Docs: Release notes for 1.9.0 (#12911)
apache/iceberg: site/docs/releases.md

</dependencies>
```

### 1.9.0 release

Apache Iceberg 1.9.0 was released on April 28, 2025.

The 1.9.0 release contains bug fixes and new features. For full release notes visit [Github](https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.9.0)

* Deprecation / End of Support
    - Spark: Remove Spark 3.3 support ([\#12279](https://github.com/apache/iceberg/pull/12279))
    - Build: Remove Hadoop 2 ([\#12348](https://github.com/apache/iceberg/pull/12348))
* Spec
    - Spec: Support geo type ([\#10981](https://github.com/apache/iceberg/pull/10981))
    - Spec: Allow Equality Deletes with Row Lineage and Define Behavior ([\#12230](https://github.com/apache/iceberg/pull/12230))

>     - Spec: Add implementation note on current-snapshot-id ([\#12334](https://github.com/apache/iceberg/pull/12334))

Reviewer: ```suggestion
    - Spec: Add implementation note on `current-snapshot-id` ([\#12334](https://github.com/apache/iceberg/pull/12334))
```

    - Implement Variant#toString ([\#12531](https://github.com/apache/iceberg/pull/12531))
* Core
    - Add partition stats writer and reader ([\#11216](https://github.com/apache/iceberg/pull/11216))
    - Auth Manager API enablement ([\#12197](https://github.com/apache/iceberg/pull/12197))
    - Add InternalData read and write builders ([\#12060](https://github.com/apache/iceberg/pull/12060))
    - Enable row lineage for all v3 tables ([\#12593](https://github.com/apache/iceberg/pull/12593))
    - FileRewritePlanner implementation ([\#12493](https://github.com/apache/iceberg/pull/12493))
    - Interface changes for separating rewrite planner and runner ([\#12306](https://github.com/apache/iceberg/pull/12306))
    - Add variant type support to utils and visitors ([\#11831](https://github.com/apache/iceberg/pull/11831))
    - Add Variant logical type for Avro ([\#12238](https://github.com/apache/iceberg/pull/12238))
    - Add variant readers and writers ([\#12457](https://github.com/apache/iceberg/pull/12457))
    - Remove namespace/table/view HEAD endpoints from defaults ([\#12351](https://github.com/apache/iceberg/pull/12351))

>     - Support timestamp(9) and unknown types ([\#12455](https://github.com/apache/iceberg/pull/12455))

Reviewer: ```suggestion
    - Support nanosecond timestamps and unknown types ([\#12455](https://github.com/apache/iceberg/pull/12455))
```

>     - Write null for current-snapshot-id for V3+ ([\#12335](https://github.com/apache/iceberg/pull/12335))

Reviewer: ```suggestion
    - Write null for `current-snapshot-id` for V3+ ([\#12335](https://github.com/apache/iceberg/pull/12335))
```

    - Apply correct metric configs in GenericAppenderFactory ([\#12366](https://github.com/apache/iceberg/pull/12366))
    - Close FileIO instance in JdbcCatalog ([\#12540](https://github.com/apache/iceberg/pull/12540))
    - Add view-override catalog property ([\#12534](https://github.com/apache/iceberg/pull/12534))

>     - Use InternalData with Avro for readers. ([\#12476](https://github.com/apache/iceberg/pull/12476))

Reviewer 2: ```suggestion
    - Use InternalData with Avro for readers ([\#12476](https://github.com/apache/iceberg/pull/12476))
```

    - Fix missing data when writing unknown ([\#12581](https://github.com/apache/iceberg/pull/12581))
    - Bulk deletion in RemoveSnapshots ([\#11837](https://github.com/apache/iceberg/pull/11837))
    - Add update event for rewrite manifests ([\#12627](https://github.com/apache/iceberg/pull/12627))
    - Add commit metrics for rewriting manifests ([\#12630](https://github.com/apache/iceberg/pull/12630))
    - Add geometry and geography types support ([\#12346](https://github.com/apache/iceberg/pull/12346))
    - Add MetricsReporter for SnapshotManager ([\#12665](https://github.com/apache/iceberg/pull/12665))
* Parquet
    - Implement Variant readers ([\#12139](https://github.com/apache/iceberg/pull/12139))
    - Implement Variant writers ([\#12323](https://github.com/apache/iceberg/pull/12323))

>     - Support unknown and timestamp(9) in internal model and generics ([\#12463](https://github.com/apache/iceberg/pull/12463))

Reviewer: ```suggestion
    - Support unknown and nanosecond timestamps in internal model and generics ([\#12463](https://github.com/apache/iceberg/pull/12463))
```

    - Implement Variant metrics ([\#12496](https://github.com/apache/iceberg/pull/12496))
* ORC

>     - Support timestamp(9), variant, and unknown in generics ([\#12567](https://github.com/apache/iceberg/pull/12567))

Reviewer: ```suggestion
    - Support nanosecond timestamps, variant, and unknown in generics ([\#12567](https://github.com/apache/iceberg/pull/12567))
```

* AWS
    - Integrate S3 analytics accelerator library ([\#12299](https://github.com/apache/iceberg/pull/12299))
* Spark
    - Rewrite V2 deletes to V3 DVs ([\#12250](https://github.com/apache/iceberg/pull/12250))
    - Detect dangling DVs properly ([\#12270](https://github.com/apache/iceberg/pull/12270))

>     - Add _row_id and _last_updated_sequence_number readers ([\#12836](https://github.com/apache/iceberg/pull/12836))

Reviewer: ```suggestion
    - Add `_row_id` and `_last_updated_sequence_number` readers ([\#12836](https://github.com/apache/iceberg/pull/12836))
```
